Generate commit messages using unstaged changes if no staged changes exist

hassoncs opened the pull request:
This change enhances the commit message generation logic to first check for staged changes. If no staged changes are found, it will then attempt to generate the commit message based on unstaged changes.

This provides a more flexible user experience, allowing users to generate commit messages even when they haven't explicitly staged their changes.## Context

Commit:
feat(commit-message): Generate commit messages based on unstaged changes

Adds the ability to generate commit messages from unstaged changes if no staged changes are present. This improves the user experience by providing a fallback mechanism for commit message generation.

- **CommitMessageProvider**: Modified to first check for staged changes, and if none exist, then check for unstaged changes. A new information message is displayed to the user when generating from unstaged changes.
- **GitExtensionService**: Updated `gatherChanges` and `getCommitContext` methods to accept an `options` object to specify whether to look for staged or unstaged changes. This allows for more flexible Git operations.
- **i18n**: Added new translation keys for the "generating from unstaged changes" message across all supported languages.
- **Tests**: Updated `CommitMessageProvider.test.ts` and `GitExtensionService.spec.ts` to reflect the new behavior and test cases for both staged and unstaged change scenarios.

chrarnoldus commented on src/services/commit-message/CommitMessageProvider.ts:
```diff
@@ -66,17 +67,21 @@ export class CommitMessageProvider {
+					if (!changes || changes.length === 0) {
+						staged = false
+						changes = await this.gitService.gatherChanges({ staged })
+						if (!changes || changes.length === 0) {
```
```suggestion
						if (!changes) {
```

Isn't the second check superfluous?

hassoncs replied:
It's a empty array vs null thing, but I think I could clean this up by updating `gitService.gatherChanges` to always be empty array

Commit:
refactor(git): Handle empty arrays instead of null for git changes

The `gatherChanges` method in `GitExtensionService` now consistently returns an empty array `[]` instead of `null` when no changes are found or an error occurs.

This change simplifies the logic in `CommitMessageProvider` by removing redundant `null` checks and allowing for more straightforward handling of empty results. Test cases have been updated to reflect this new behavior, ensuring consistency and robustness.

Commit:
Update strings
